src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -2012,24 +2011,45 @@ private static void ImportKnownTypeAttributes(Type? type, Dictionary<Type, Type>
                     }
                 }
 
-                //For Json we need to add KeyValuePair<K,T> to KnownTypes if the UnderLyingType is a Dictionary<K,T>
-                try
+                AppContext.TryGetSwitch("Switch.System.Runtime.Serialization.DataContracts.Auto_Import_KVP", out bool autoImportKVP);
+                if (autoImportKVP)
                 {
-                    if (DataContract.GetDataContract(type) is CollectionDataContract collectionDataContract && collectionDataContract.IsDictionary &&
-                        collectionDataContract.ItemType.GetGenericTypeDefinition() == Globals.TypeOfKeyValue)
+                    //For Json we need to add KeyValuePair<K,T> to KnownTypes if the UnderLyingType is a Dictionary<K,T>
+                    //
+                    // ^^^ Years later I think this is referring to how we treat dictionaries as collections of the
+                    // 'KeyValue' struct we have internally. But Json for some reason needed contracts for KeyValuePair<K,T>
+                    // as well. Possibly related to 'UseSimpleDictionaryFormat'? So we would import both KeyValue and KVP
+                    // contracts and be happy.
+                    // Years later, we've had and gotten rid of a KeyValuePair adapter, and I'm not sure I see a case where
+                    // DCJS is failing with either dictionary format without this code. This seems extraneous at this
+                    // point. Perhaps its there to bridge a potential mis-match between producer and consumer?
+                    //
+                    //
+                    // Or, if I put more faith in the 'catch' comment - this was a case where DCSJ might have used
+                    // an internal KV struct because KVP was missing [Serializable], but regular DCS was using a
+                    // KVPAdapter class? And now KVP does have [Serializable] so we don't need either of those workarounds
+                    // anymore. We already dropped KVPAdapter in this PR. We can probably drop this workaround as well.
+                    // This seems the more likely explaination for this block.
```

Comment:
Maybe add mention that having this here causes an extra Xsd schema entry when exporting the data contract as justification for removing this by default?